get-projects.py

```diff
@@ -37,18 +39,28 @@ def get_keystone_url(os_auth_url, path):
     return urlunparse((url[0], url[1], path, url[3], url[4], url[5]))
 
 
-def get_unscoped_token(os_auth_url, access_token):
+def get_unscoped_token(os_auth_url, access_token, method='oidc'):
     url = get_keystone_url(
         os_auth_url,
-        "/v3/OS-FEDERATION/identity_providers/egi.eu/protocols/oidc/auth")
+        "/v3/OS-FEDERATION/identity_providers/egi.eu/protocols/%s/auth" %
+        method)
     r = requests.post(url,
                       headers={'Authorization': 'Bearer %s' % access_token})
-    return r.headers['X-Subject-Token']
+    if r.status_code != requests.codes.created:
+        if method != 'openid':
+            return get_unscoped_token(os_auth_url, access_token, 'openid')
```

Comment:
This recursive call feels a bit weird to me and makes the function harder to test (if we were doing unit testing). 

Could you raise here an exception and do the re-try with another method outside?

Reply:
Done.